ext/sdk/resources/sdk-root/shell/src/resource-templates/ts/scaffolder.ts

```diff
@@ -0,0 +1,181 @@
+          this.fsService.writeFile(relativePath('client/tsconfig.json'), getClientTSConfig()),
+        )));
+
+    // Server Script
+    promises.push(this.fsService.mkdirp(relativePath('server'))
+      .then(() => concurrently(
+          this.fsService.writeFile(relativePath('server/server.ts'), `console.log("[${resourceName}] Server Resource Started");`),
+          this.fsService.writeFile(relativePath('server/tsconfig.json'), getServerTSConfig()),
+        )));
+
+    // Build Tools
+    promises.push(this.fsService.writeFile(relativePath('package.json'), getPackageJson(resourceName)));
+    promises.push(this.fsService.writeFile(relativePath('build.js'), getBuildJS()));
+    promises.push(this.fsService.writeFile(relativePath('.fxdkignore'), getIgnore()));
+
+    await Promise.all(promises);
+
+    await this.installModules(resourcePath);
+  }
+
+  private async installModules(path: string) {
+    return new Promise<void>((resolve, reject) => {
+      cp.exec("yarn", { cwd: path, windowsHide: true }, (err) => {
+        if (err) {
+          return reject(err);
+        }
+
+        return resolve();
+      });
+    });
+  }
+}
+
+function getIgnore(): string {
+  return `client/
+server/
+node_modules/
+package.json
+build.js
+`;
+}
+
+function getClientTSConfig(): string {
+  return `{
+  "compilerOptions": {
+    "baseUrl": ".",
+    "paths": {
+      "*": ["types/*"]
+    },
+    "outDir": "./",
+    "noImplicitAny": true,
+    "module": "commonjs",
+    "target": "es2016",
+    "lib": ["es2017"],
+    "types": ["@citizenfx/client", "@types/node"],
+    "moduleResolution": "node",
+    "resolveJsonModule": true,
+    "esModuleInterop": true,
+    "noEmit": true,
+  },
+  "include": ["./**/*", "../common/**/*"],
+  "exclude": ["**/node_modules"]
+}`;
+}
+
+function getServerTSConfig(): string {
+  return `{
+  "compilerOptions": {
+    "baseUrl": ".",
+    "paths": {
+      "*": ["types/*"]
+    },
+    "outDir": "./",
+    "noImplicitAny": true,
+    "module": "commonjs",
+    "target": "es2016",
+    "allowJs": false,
+    "lib": ["es2015", "DOM"],
+    "types": ["@citizenfx/server", "@types/node"],
+    "moduleResolution": "node",
+    "emitDecoratorMetadata": true,
+    "experimentalDecorators": true,
+    "resolveJsonModule": true,
+    "esModuleInterop": true,
+    "noEmit": true
+  },
+  "include": ["./**/*", "../common/**/*"],
+  "exclude": ["**/node_modules"]
+}`;
+}
+
+function getPackageJson(resourceName: string): string {
+  return `{
+  "name": "${resourceName}",
+  "version": "1.0.0",
+  "main": "index.js",
+  "author": "you",
+  "license": "MIT",
+  "scripts": {
+    "build": "cross-env MODE=production node build.js",
+    "watch": "cross-env MODE=development node build.js"
+  },
+  "devDependencies": {
+    "@citizenfx/client": "latest",
+    "@citizenfx/server": "latest",
+    "@types/node": "^16.3.1",
+    "cross-env": "7.0.3",
```

Comment:
Please use CLI option passing instead of adding this dep. For example `node build.js --mode=production`.

Reply:
For the reason of not wanting to pull in a process argument library to parse args, I did the simplest implementation of this suggestion via

`node build.js production` and `node build.js development`

and then just checked
`const production = process.argv[2] === 'production' ? true : false;`

Reply:
I'm still not totally happy with this solution though, and am open to suggestions

Reply:
Can just reuse this code: https://github.com/citizenfx/fivem/blob/master/ext/sdk-build/utils.js#L54-L63

Usage be like: https://github.com/citizenfx/fivem/blob/master/ext/sdk-build/build.js#L4-L5